x-pack/plugins/alerting/server/alert/create_alert_factory.test.ts

```diff
@@ -298,6 +310,34 @@ describe('createAlertFactory()', () => {
     alertFactory.alertLimit.setLimitReached(false);
     alertFactory.alertLimit.checkLimitUsage();
   });
+
+  test('trimRecovered should return longest recovered alerts', () => {
+    const alertFactory = createAlertFactory({
+      alerts: {},
+      logger,
+      maxAlerts: 2,
+    });
+
+    const recoveredAlerts = [alert1, alert2, alert3];
+    const trimmedAlerts = alertFactory.alertLimit.trimRecovered(recoveredAlerts);
+    expect(trimmedAlerts).toEqual([alert2]);
```

Comment:
Shouldn't we expect alert2 and alert1 given that the maxAlerts is 2?

Reply:
The function returns the alerts that were trimmed, instead of the recovered alerts that were not trimmed